test/integration/Test/Integration/Framework/DSL.hs

```diff
@@ -77,9 +126,111 @@ expectResponseCode
 expectResponseCode want (got, _) =
     got `shouldBe` want
 
+expectFieldEqual
+    :: (MonadIO m, MonadFail m, Show a, Eq a)
+    => Lens' s a
+    -> a
+    -> (HTTP.Status, Either RequestException s)
+    -> m ()
+expectFieldEqual getter a (_, res) = case res of
+    Left e  -> wantedSuccessButError e
+    Right s -> (view getter s) `shouldBe` a
+
+--
+-- Lenses
+--
+addressPoolGap :: HasType (ApiT AddressPoolGap) s => Lens' s Word8
+addressPoolGap =
+    lens _get _set
+  where
+    _get :: HasType (ApiT AddressPoolGap) s => s -> Word8
+    _get = getAddressPoolGap . getApiT . view typed
+    _set :: HasType (ApiT AddressPoolGap) s => (s, Word8) -> s
+    _set (s, v) = set typed (ApiT $ unsafeMkAddressPoolGap v) s
+
+balance :: HasType (ApiT WalletBalance) s => Lens' s WalletBalance
+balance =
+    lens _get _set
+  where
+    _get :: HasType (ApiT WalletBalance) s => s -> WalletBalance
+    _get = getApiT . view typed
+    _set :: HasType (ApiT WalletBalance) s => (s, WalletBalance) -> s
+    _set (s, v) = set typed (ApiT v) s
+
+balanceAvailable :: HasType (ApiT WalletBalance) s => Lens' s (Quantity "lovelace" Natural)
+balanceAvailable =
+    lens _get _set
+  where
+    _get :: HasType (ApiT WalletBalance) s => s -> Quantity "lovelace" Natural
+    _get = available . getApiT . view typed
+    _set :: HasType (ApiT WalletBalance) s => (s, Quantity "lovelace" Natural) -> s
+    _set (s, v) = set typed (ApiT $ WalletBalance {available = v, total = v }) s
+
+balanceTotal :: HasType (ApiT WalletBalance) s => Lens' s (Quantity "lovelace" Natural)
+balanceTotal =
+    lens _get _set
+  where
+    _get :: HasType (ApiT WalletBalance) s => s -> Quantity "lovelace" Natural
+    _get = total . getApiT . view typed
+    _set :: HasType (ApiT WalletBalance) s => (s, Quantity "lovelace" Natural) -> s
+    _set (s, v) = set typed (ApiT $ WalletBalance {available = v, total = v }) s
+
+delegation :: HasType (ApiT (WalletDelegation (ApiT PoolId))) s => Lens' s (WalletDelegation (ApiT PoolId))
```

Comment:
`Lens' s Text`

Reply:
Yeah, I'm having troubles with that one :)
I wanted to go for something like:
```haskell
delegation :: HasType (ApiT (WalletDelegation (ApiT PoolId))) s => Lens' s Text
delegation =
    lens _get _set
  where
    _get :: HasType (ApiT (WalletDelegation (ApiT PoolId))) s => s -> Text
    _get = case d of
	           NotDelegating -> "NotDelegating"
	           Delegating (ApiT (PoolId p)) -> (T.append "Delegating " p)
           where
	           d = getApiT . view typed
    _set :: HasType (ApiT (WalletDelegation (ApiT PoolId))) s => (s, Text) -> s
    _set (s, v) = set typed (ApiT init) s
		          where
		           init = if (v == "NotDelegating")
			            then NotDelegating
			            else Delegating (ApiT (PoolId (T.drop 11 v)))
```
but obviously that does not type check...